File: javascript/soma_matrizes.js
const m = parseInt(prompt("Quantas linhas vai ter cada matriz?"));
const n = parseInt(prompt("Quantas colunas vai ter cada matriz?"));

let A = [], B = [], matrizSoma = [];

for(i = 0; i < m; i++) {
    A.push([]);
    B.push([]);
    matrizSoma.push([]);
}

alert("Digite os valores da matriz B:");
for(i = 0; i < m; i++) {
    for(j = 0; j < n; j++) {
        B[i][j] = parseInt(prompt("Elemento [" + i + "," + j + "]:"));
    }
}

for(i = 0; i < m; i++) {
    for(j = 0; j < n; j++) {
        matrizSoma[i][j] = A[i][j] + B[i][j];
    }
}

console.log("MATRIZ SOMA:");
for(i = 0; i < m; i++) {
    let aux = "";
    for(j = 0; j < n; j++) {
        aux += matrizSoma[i][j] + " ";
    }
    console.log(aux);
}

File: python/soma_matrizes.py
def criar_matriz(num_linhas: int, num_colunas: int) -> list[list[int]]:
    m = []

    for i in range(num_linhas):
        m.append([0] * num_colunas)

    return m


def ler_matriz(matriz: list[list[int]]) -> None:
    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))


def imprimir_matriz(matriz: list[list[int]]) -> None:
    for i in range(len(matriz)):
        for j in range(len(matriz[0])):
            print(matriz[i][j], end=" ")
        print()


m = int(input("Quantas linhas vai ter cada matriz? "))
n = int(input("Quantas colunas vai ter cada matriz? "))

print("Digite os valores da matriz A:")
a = criar_matriz(m, n)
ler_matriz(a)

print("Digite os valores da matriz B:")
b = criar_matriz(m, n)
ler_matriz(b)

c = criar_matriz(m, n)
for i in range(m):
    for j in range(n):
        c[i][j] = a[i][j] + b[i][j]

print("MATRIZ SOMA:")
imprimir_matriz(c)
